docs/utilities/idempotency.md

```diff
@@ -0,0 +1,797 @@
+        runtime       = "nodejs18.x"
+        handler       = "index.handler"
+        filename      = "lambda.zip"
+    }
+
+    resource "aws_iam_role" "IdempotencyFunctionRole" {
+        name = "IdempotencyFunctionRole"
+
+        assume_role_policy = jsonencode({
+            Version = "2012-10-17"
+            Statement = [
+            {
+                Sid    = ""
+                Effect = "Allow"
+                Principal = {
+                Service = "lambda.amazonaws.com"
+                }
+                Action = "sts:AssumeRole"
+            },
+            ]
+        })
+    }
+
+    resource "aws_iam_policy" "LambdaDynamoDBPolicy" {
+        name        = "LambdaDynamoDBPolicy"
+        description = "IAM policy for Lambda function to access DynamoDB"
+        policy = jsonencode({
+            Version = "2012-10-17"
+            Statement = [
+            {
+                Sid    = "AllowDynamodbReadWrite"
+                Effect = "Allow"
+                Action = [
+                "dynamodb:PutItem",
+                "dynamodb:GetItem",
+                "dynamodb:UpdateItem",
+                "dynamodb:DeleteItem",
+                ]
+                Resource = aws_dynamodb_table.IdempotencyTable.arn
+            },
+            ]
+        })
+    }
+
+    resource "aws_iam_role_policy_attachment" "IdempotencyFunctionRoleAttachment" {
+        role       = aws_iam_role.IdempotencyFunctionRole.name
+        policy_arn = aws_iam_policy.LambdaDynamoDBPolicy.arn
+    }
+    ```
+
+???+ warning "Warning: Large responses with DynamoDB persistence layer"
+    When using this utility with DynamoDB, your function's responses must be [smaller than 400KB](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Limits.html#limits-items){target="_blank"}.
+
+    Larger items cannot be written to DynamoDB and will cause exceptions.
+
+???+ info "Info: DynamoDB"
+    Each function invocation will generally make 2 requests to DynamoDB. If the
+    result returned by your Lambda is less than 1kb, you can expect 2 WCUs per invocation. For retried invocations, you will
+    see 1WCU and 1RCU. Review the [DynamoDB pricing documentation](https://aws.amazon.com/dynamodb/pricing/){target="_blank"} to
+    estimate the cost.
+
+### MakeIdempotent function wrapper
+
+You can quickly start by initializing the `DynamoDBPersistenceLayer` class and using it with the `makeIdempotent` function wrapper on your Lambda handler.
+
+???+ note
+    In this example, the entire Lambda handler is treated as a single idempotent operation. If your Lambda handler can cause multiple side effects, or you're only interested in making a specific logic idempotent, use the `makeIdempotent` high-order function only on the function that needs to be idempotent.
+
+!!! tip "See [Choosing a payload subset for idempotency](#choosing-a-payload-subset-for-idempotency) for more elaborate use cases."
+
+=== "index.ts"
+
+    ```typescript hl_lines="2-3 21 35-38"
```

Comment:
It was intentional, the last line is the closing `)` of the wrapper and I wanted to call attention to the fact that it needs to be closed:

![Screenshot 2023-07-10 at 18 12 39](https://github.com/aws-powertools/powertools-lambda-typescript/assets/7353869/bcbeba2d-3afd-47a0-91a2-5cc4d5a26ecd)

I'm worried that removing it would result in people selecting only till it's highlighted & copy-pasting a snippet that is incomplete (aka missing the last `)`).

If it's overkill I can remove it though.